UndertaleModTool/Scripts/Community Scripts/TouchControlsEnabler.csx

```diff
@@ -0,0 +1,452 @@
+    MarginTop = 0,
+    MarginBottom = 11,
+};
+spr_analog_scale.Textures.Add(new UndertaleSprite.TextureEntry() { Texture = pg_analog_scale });
+Data.Sprites.Add(spr_analog_scale);
+
+var spr_analog_type = new UndertaleSprite()
+{
+    Name = Data.Strings.MakeString("spr_analog_type"),
+    Width = 72,
+    Height = 12,
+    MarginLeft = 0,
+    MarginRight = 71,
+    MarginTop = 0,
+    MarginBottom = 11,
+};
+spr_analog_type.Textures.Add(new UndertaleSprite.TextureEntry() { Texture = pg_analog_type });
+Data.Sprites.Add(spr_analog_type);
+
+var spr_reset_config = new UndertaleSprite()
+{
+    Name = Data.Strings.MakeString("spr_reset_config"),
+    Width = 79,
+    Height = 12,
+    MarginLeft = 0,
+    MarginRight = 78,
+    MarginTop = 0,
+    MarginBottom = 11,
+};
+spr_reset_config.Textures.Add(new UndertaleSprite.TextureEntry() { Texture = pg_reset_config });
+Data.Sprites.Add(spr_reset_config);
+
+var spr_controls_opacity = new UndertaleSprite()
+{
+    Name = Data.Strings.MakeString("spr_controls_opacity"),
+    Width = 107,
+    Height = 13,
+    MarginLeft = 0,
+    MarginRight = 106,
+    MarginTop = 0,
+    MarginBottom = 12,
+};
+spr_controls_opacity.Textures.Add(new UndertaleSprite.TextureEntry() { Texture = pg_controls_opacity });
+Data.Sprites.Add(spr_controls_opacity);
+
+var spr_arrow_leftright = new UndertaleSprite()
+{
+    Name = Data.Strings.MakeString("spr_arrow_leftright"),
+    Width = 41,
+    Height = 9,
+    MarginLeft = 0,
+    MarginRight = 40,
+    MarginTop = 0,
+    MarginBottom = 8,
+};
+spr_arrow_leftright.Textures.Add(new UndertaleSprite.TextureEntry() { Texture = pg_arrow_leftright });
+Data.Sprites.Add(spr_arrow_leftright);
+
+var spr_black = new UndertaleSprite()
+{
+    Name = Data.Strings.MakeString("spr_black"),
+    Width = 640,
+    Height = 480,
+    MarginLeft = 0,
+    MarginRight = 639,
+    MarginTop = 0,
+    MarginBottom = 479,
+};
+spr_black.Textures.Add(new UndertaleSprite.TextureEntry() { Texture = pg_black });
+Data.Sprites.Add(spr_black);
+
+ImportGMLFile(dataPath+"/gml_Object_obj_mobilecontrols_Create_0.gml", true, false, true);
+ImportGMLFile(dataPath+"/gml_Object_obj_mobilecontrols_Draw_64.gml", true, false, true);
+ImportGMLFile(dataPath+"/gml_Object_obj_mobilecontrols_Other_4.gml", true, false, true);
+Data.Scripts.Add(new UndertaleScript() { Name = Data.Strings.MakeString("scr_add_keys"), Code = Data.Code.ByName("gml_Object_obj_mobilecontrols_Other_4") });
+ImportGMLFile(dataPath+"/gml_Object_obj_mobilecontrols_Step_0.gml", true, false, true);
+
+if (Data?.GeneralInfo?.DisplayName?.Content.ToLower() == "deltarune chapter 1 & 2")
+{
+    Data.Code.ByName("gml_Object_obj_gamecontroller_Create_0").AppendGML("instance_create(0, 0, obj_mobilecontrols);", Data);
+}
+else if (Data?.GeneralInfo?.DisplayName?.Content.ToLower() == "deltarune chapter 1&2")
+{
+    Data.Code.ByName("gml_Object_obj_gamecontroller_Create_0").AppendGML("instance_create(0, 0, obj_mobilecontrols);", Data);
+} 
```

Comment:
whats the deal with these?

Reply:
yeah i meant why do the blocks have the exact same code. should've been more clearer on that.

Reply:
any other comment on this or should I mark it as resolved?

Reply:
if it already looks something like
```cs
if (displayName == "deltarune chapter 1 & 2" || displayName == "deltarune chapter 1&2")
{
  Data.Code.ByName("gml_Object_obj_gamecontroller_Create_0").AppendGML("instance_create(0, 0, obj_mobilecontrols);", Data);
}
else if (displayName == "undertale")
{
  Data.Code.ByName("gml_Object_obj_time_Create_0").AppendGML("instance_create(0, 0, obj_mobilecontrols);", Data);
}
```

for you locally, then feel  free to resolve.

Reply:
yep, it looks exactly like this.